### PR TITLE
doc(#318): add documentation about WinError 995

### PR DESCRIPTION
**Pull request by Bogay**
## Description

Add some description related to #318 so that others may find the solution on documentation.

## Checklist

- [x] Add test cases to all the changes you introduce
- [x] Run `./script/format` and `./script/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -73,3 +73,16 @@ If you are using conventional commits in your git history, then you could swap o
...
+## How to handle revert commits?
+
+```sh
+git revert --no-commit <SHA>
+git commit -m "revert: foo bar"
+```
+
+## I got `Exception [WinError 995] The I/O operation ...` error
+
+This error is cause by a python bug on windows and it should be fixed in some newer version. (e.g. 3.8.7) So if you encounter this error. Upgrade python version may solve this.
```
Comment (Lee-W) [inline docs/faq.md:86]: I might reword it to

```
This error was caused by a Python bug on Windows. It's been fixed and tested on Python 3.8.7. Upgrade your Python version might solve this issue.
```

**Review by Lee-W — COMMENTED** [review]
Other than the minor documentation reword. Everything looks good. Thanks!

File: docs/faq.md (hunk 1 above)
Comment (Bogay) [reply docs/faq.md:86]: But 3.8.7 is not the only version that include this fix. According to Python's changelog, [3.8.6rc1](https://docs.python.org/3.8/whatsnew/changelog.html#python-3-8-6-release-candidate-1) and [3.9.0rc2](https://docs.python.org/3.9/whatsnew/changelog.html#python-3-9-0-release-candidate-2) should be the accurate versions first contain https://github.com/python/cpython/pull/22017.  
I am afraid that just say it has been fixed and tested on Python 3.8.7 might mislead readers this error would not happened on Python 3.9. Should I write these information to the doc?

File: docs/README.md [hunk 2]
```diff
@@ -175,23 +175,11 @@ See [Third-Party Commitizen Templates](third-party-commitizen.md).
...
-
-See more discussion in issue [#142](https://github.com/commitizen-tools/commitizen/issues/142) and [#36](https://github.com/commitizen-tools/commitizen/issues/36)
-
-### How to handle revert commits?
-
-```sh
-git revert --no-commit <SHA>
-git commit -m "revert: foo bar"
-```
+See [FAQ - commitizen](https://commitizen-tools.github.io/commitizen/faq/).
```
Comment (Bogay) [inline docs/README.md:178]: @Lee-W Should I replace the URL with filename? It seems that filenames will be rendered as URL in generated documentation.
For example, `contributing.md` will turn into https://commitizen-tools.github.io/commitizen/contributing/

File: docs/faq.md (hunk 1 above)
Comment (Lee-W) [reply docs/faq.md:86]: Yep, that would definitely be super helpful! Thanks for digging into it

File: docs/README.md (hunk 2 above)
Comment (Lee-W) [reply docs/README.md:178]: Yep, that would be even much better. 💯 didn't notice this one.

Comment (Bogay) [reply docs/README.md:178]: Should I also replace those URLs inside other files? But I am not sure how to reference files not in the same directory.  
For example, near the end of this line, `docs/tutorials/gitlab_ci.md` refers to `docs/bump.md`.

https://github.com/commitizen-tools/commitizen/blob/061f47c1434f2df2da49579ff92e489001ecd3ec/docs/tutorials/gitlab_ci.md#L13

There seems to exist 2 valid path? `../bump.md` and `/docs/bump.md`, I am not sure which one is better and whether they will work in the final documentation.

Comment (Lee-W) [reply docs/README.md:178]: I would prefer `../bump.md` in this case. If my understanding is correct, when deployed to gh pages `/docs/bump.md` becomes `https://commitizen-tools.github.io/docs/bump.md`

Comment (Lee-W) [reply docs/README.md:178]: And yes, if you found any other problem, it would be better if you could fix them as well

Comment (Bogay) [reply docs/README.md]: I have updated all the links I can find in documentation, and I found that absolute path like `/docs/bump.md` is not support by MkDocs, the documentation is [here](https://www.mkdocs.org/user-guide/writing-your-docs/#linking-to-pages).

File: docs/README.md [hunk 3]
```diff
@@ -175,23 +175,11 @@ See [Third-Party Commitizen Templates](third-party-commitizen.md).
 
 ## FAQ
```
Comment (Lee-W) [inline docs/README.md:176]: I think we could remove this section. as the content has been moved to FAQ page

File: docs/config.md [hunk 4]
```diff
@@ -134,9 +134,9 @@ commitizen:
 | `customize`      | `dict` | `None`                      | **This is only supported when config through `toml`.** Custom rules for committing and bumping. [See more](customization)                                     |
 | `use_shortcuts`  | `bool` | `false`                     | If enabled, commitizen will show keyboard shortcuts when selecting from a list. Define a `key` for each of your choices to set the key. [See more](shortcuts) |
 
-[version_files]: https://commitizen-tools.github.io/commitizen/bump/#version_files
```
Comment (Lee-W) [inline docs/config.md:137]: This syntax doesn't seem to work. Could you please check again?

File: docs/tutorials/writing_commits.md [hunk 5]
```diff
@@ -38,5 +38,5 @@ understand what happened.
 | `fix(commands): bump error when no user provided` | `fix: stuff` |
 | `feat: add new commit command` | `feat: commit command introduced` |
 
-[customization]: https://commitizen-tools.github.io/commitizen/customization/
+[customization]: customization.md
```
Comment (Lee-W) [inline docs/tutorials/writing_commits.md:41]: This does not work either

**Review by Lee-W — CHANGES_REQUESTED** [review]
Hi, sorry for the late review. As PyCon TW approaches, I get less time to devote to this project. Here are some errors I found on the changes.

You could test out through

```sh
poetry run mkdocs build
poetry run mkdocs serve
```

Would be great if you could add these commands to "how to contribute" so that others can have a better idea on how to test documentation changes

**Comment by Lee-W** [comment]
Also, you might need to rebase the latest master branch into this branch to pass the test fail

File: docs/README.md (hunk 3 above)
Comment (Bogay) [reply docs/README.md:176]: OK, should I also remove "Third-Party Commitizen Templates" and "Contributing" sections?
The content is also in other pages and only one link in README.

**Comment by Lee-W** [comment]
I still see the warning here.
image

Maybe you could try the following command to fix it

```sh
git remote add upstream git@github.com:commitizen-tools/commitizen.git
git fetch upstream 
git rebase -i remotes/upstream/main
```

**Comment by Bogay** [comment]
@Lee-W sorry for lately fix and thank you for helping me to rebase the upstream. I have finished those changes.
But the test coverage seems to decrease a little, I have no idea about that. Did I break something?
